In [133]:
import plotly.plotly as py
from plotly.graph_objs import *
import pandas as pd

In [134]:
%matplotlib inline

In [135]:
df_2009 = pd.read_pickle('2009_full_score_projects_final.pkl')

In [136]:
df_2009.head()

ssc1  ssc2  ssc3  ssc4_1  ssc4_2  ssc4_3  ssc4_4  ssc5_1  ssc5_2  \
project_id                                                                     
1000000143     1     5     0       6       0       0       2       0       1   
1000000174     0     0     0       6       1       0       0       0       1   
1000000194     1     5     0       0       1       3       2       0       0   
1000000199     1     0     0       0       1       3       0       0       1   
1000000222     1     5     0       6       1       3       2       0       0   

            ssc6_1        ...          country   leed_version  \
project_id                ...                                   
1000000143       1        ...               US  LEED-NC v2009   
1000000174       0        ...               US  LEED-NC v2009   
1000000194       0        ...               US  LEED-NC v2009   
1000000199       0        ...               US  LEED-NC v2009   
1000000222       0        ...               US  LEED-NC v2009   

            points_achieved  certification_level  certification_date  \
project_id                                                             
1000000143               44            Certified          2013-09-20   
1000000174               65                 Gold          2010-12-02   
1000000194               44            Certified          2011-08-23   
1000000199               51               Silver          2012-03-26   
1000000222               70                 Gold          2014-01-17   

                                 owner_types  gross_square_foot  \
project_id                                                        
1000000143             Government Use: State              86580   
1000000174  Educational: University, Private              11806   
1000000194         Corporate: Privately Held               3472   
1000000199       Government Use: Local, City              48732   
1000000222   Educational: University, Public             212777   

            total_property_area                                project_types  \
project_id                                                                     
1000000143                53863                      Health Care: Outpatient   
1000000174                66000      Core Learning Space: College/University   
1000000194                10905                            Commercial Office   
1000000199               555823  Public Order and Safety: Other Public Order   
1000000222               108563      Core Learning Space: College/University   

            registration_date  
project_id                     
1000000143         2009-05-07  
1000000174         2009-05-10  
1000000194         2009-05-12  
1000000199         2009-05-13  
1000000222         2009-05-14  

[5 rows x 67 columns]

In [137]:
df_2009['year'] = df_2009['certification_date'].apply(lambda x: x.strftime('%d%m%Y'))

In [138]:
df_2009['year_1'] = df_2009['year'].apply(lambda x: x[-4:])

In [139]:
df_2009.head()

ssc1  ssc2  ssc3  ssc4_1  ssc4_2  ssc4_3  ssc4_4  ssc5_1  ssc5_2  \
project_id                                                                     
1000000143     1     5     0       6       0       0       2       0       1   
1000000174     0     0     0       6       1       0       0       0       1   
1000000194     1     5     0       0       1       3       2       0       0   
1000000199     1     0     0       0       1       3       0       0       1   
1000000222     1     5     0       6       1       3       2       0       0   

            ssc6_1   ...    points_achieved  certification_level  \
project_id           ...                                           
1000000143       1   ...                 44            Certified   
1000000174       0   ...                 65                 Gold   
1000000194       0   ...                 44            Certified   
1000000199       0   ...                 51               Silver   
1000000222       0   ...                 70                 Gold   

            certification_date                       owner_types  \
project_id                                                         
1000000143          2013-09-20             Government Use: State   
1000000174          2010-12-02  Educational: University, Private   
1000000194          2011-08-23         Corporate: Privately Held   
1000000199          2012-03-26       Government Use: Local, City   
1000000222          2014-01-17   Educational: University, Public   

            gross_square_foot  total_property_area  \
project_id                                           
1000000143              86580                53863   
1000000174              11806                66000   
1000000194               3472                10905   
1000000199              48732               555823   
1000000222             212777               108563   

                                          project_types  registration_date  \
project_id                                                                   
1000000143                      Health Care: Outpatient         2009-05-07   
1000000174      Core Learning Space: College/University         2009-05-10   
1000000194                            Commercial Office         2009-05-12   
1000000199  Public Order and Safety: Other Public Order         2009-05-13   
1000000222      Core Learning Space: College/University         2009-05-14   

                year  year_1  
project_id                    
1000000143  20092013    2013  
1000000174  02122010    2010  
1000000194  23082011    2011  
1000000199  26032012    2012  
1000000222  17012014    2014  

[5 rows x 69 columns]

In [140]:
df_2014 = df_2009[df_2009['year_1']=='2014']

In [141]:
df_2013 = df_2009[df_2009['year_1']=='2013']
df_2012 = df_2009[df_2009['year_1']=='2012']
df_2011 = df_2009[df_2009['year_1']=='2011']
df_2010 = df_2009[df_2009['year_1']=='2010']

In [142]:
df_2015 = df_2009[df_2009['year_1']=='2015']

In [143]:
df_2015_trends = df_2015.describe()
df_2014_trends = df_2014.describe()
df_2013_trends = df_2013.describe()
df_2012_trends = df_2012.describe()
df_2011_trends = df_2011.describe()

In [144]:
df_2015_trends.drop('gross_square_foot', axis=1, inplace=True)

In [145]:
df_2015_trends.drop('total_property_area', axis=1, inplace=True)

In [146]:
df_2014_trends.drop('gross_square_foot', axis=1, inplace=True)
df_2014_trends.drop('total_property_area', axis=1, inplace=True)
df_2013_trends.drop('gross_square_foot', axis=1, inplace=True)
df_2013_trends.drop('total_property_area', axis=1, inplace=True)
df_2012_trends.drop('gross_square_foot', axis=1, inplace=True)
df_2012_trends.drop('total_property_area', axis=1, inplace=True)
df_2011_trends.drop('gross_square_foot', axis=1, inplace=True)
df_2011_trends.drop('total_property_area', axis=1, inplace=True)

In [147]:
df_2015_mean = df_2015_trends.iloc[[1]]
df_2014_mean = df_2014_trends.iloc[[1]]
df_2013_mean = df_2013_trends.iloc[[1]]
df_2012_mean = df_2012_trends.iloc[[1]]
df_2011_mean = df_2011_trends.iloc[[1]]

In [148]:
df_2015_mean = df_2015_mean.rename(index={'mean': '2015'})
df_2014_mean = df_2014_mean.rename(index={'mean': '2014'})
df_2013_mean = df_2013_mean.rename(index={'mean': '2013'})
df_2012_mean = df_2012_mean.rename(index={'mean': '2012'})
df_2011_mean = df_2011_mean.rename(index={'mean': '2011'})



In [149]:
frames = [df_2011_mean, df_2012_mean, df_2013_mean, df_2014_mean, df_2015_mean]
result = pd.concat(frames)

In [150]:
result.head()

ssc1      ssc2      ssc3    ssc4_1    ssc4_2    ssc4_3    ssc4_4  \
2011  0.847826  3.260870  0.173913  3.652174  0.550725  2.565217  1.246377   
2012  0.829832  2.888655  0.203782  3.063025  0.605042  2.357143  1.285714   
2013  0.841830  2.862745  0.179085  3.200000  0.661438  2.439216  1.427451   
2014  0.895294  3.088235  0.212941  3.698824  0.677647  2.361176  1.470588   
2015  0.881679  3.034351  0.232824  3.778626  0.599237  2.255725  1.564885   

        ssc5_1    ssc5_2    ssc6_1       ...           eqc7_2    eqc8_1  \
2011  0.159420  0.565217  0.434783       ...         0.659420  0.231884   
2012  0.186975  0.621849  0.451681       ...         0.638655  0.212185   
2013  0.183007  0.667974  0.439216       ...         0.631373  0.188235   
2014  0.148235  0.658824  0.416471       ...         0.585882  0.147059   
2015  0.190840  0.679389  0.412214       ...         0.503817  0.175573   

        eqc8_2      idc1      idc2    extra1    extra2    extra3    extra4  \
2011  0.485507  3.065217  0.963768  0.558824  0.607692  0.720339  0.610526   
2012  0.441176  2.920168  0.991597  0.536481  0.676923  0.648585  0.589595   
2013  0.376471  2.952941  0.998693  0.500000  0.610054  0.643388  0.578207   
2014  0.360000  2.961176  0.996471  0.503571  0.597033  0.645380  0.582456   
2015  0.377863  2.977099  0.996183  0.469231  0.596078  0.646288  0.531250   

      points_achieved  
2011        57.253623  
2012        56.697479  
2013        56.433987  
2014        55.829412  
2015        54.450382  

[5 rows x 53 columns]

In [151]:
result.rename(columns=lambda x: x.replace('_', '.'), inplace=True)

In [152]:
result.rename(columns=lambda x: x.upper(), inplace=True)

In [153]:
result.rename(columns=lambda x: x.replace('C', 'c'), inplace=True)

In [154]:
result.rename(columns=lambda x: x.replace('EXTRA', 'Extra '), inplace=True)

In [155]:
result.rename(columns=lambda x: x.replace('POINTS.AcHIEVED', 'Total Points'), inplace=True)

In [156]:
result

SSc1      SSc2      SSc3    SSc4.1    SSc4.2    SSc4.3    SSc4.4  \
2011  0.847826  3.260870  0.173913  3.652174  0.550725  2.565217  1.246377   
2012  0.829832  2.888655  0.203782  3.063025  0.605042  2.357143  1.285714   
2013  0.841830  2.862745  0.179085  3.200000  0.661438  2.439216  1.427451   
2014  0.895294  3.088235  0.212941  3.698824  0.677647  2.361176  1.470588   
2015  0.881679  3.034351  0.232824  3.778626  0.599237  2.255725  1.564885   

        SSc5.1    SSc5.2    SSc6.1      ...         EQc7.2    EQc8.1  \
2011  0.159420  0.565217  0.434783      ...       0.659420  0.231884   
2012  0.186975  0.621849  0.451681      ...       0.638655  0.212185   
2013  0.183007  0.667974  0.439216      ...       0.631373  0.188235   
2014  0.148235  0.658824  0.416471      ...       0.585882  0.147059   
2015  0.190840  0.679389  0.412214      ...       0.503817  0.175573   

        EQc8.2      IDc1      IDc2   Extra 1   Extra 2   Extra 3   Extra 4  \
2011  0.485507  3.065217  0.963768  0.558824  0.607692  0.720339  0.610526   
2012  0.441176  2.920168  0.991597  0.536481  0.676923  0.648585  0.589595   
2013  0.376471  2.952941  0.998693  0.500000  0.610054  0.643388  0.578207   
2014  0.360000  2.961176  0.996471  0.503571  0.597033  0.645380  0.582456   
2015  0.377863  2.977099  0.996183  0.469231  0.596078  0.646288  0.531250   

      Total Points  
2011     57.253623  
2012     56.697479  
2013     56.433987  
2014     55.829412  
2015     54.450382  

[5 rows x 53 columns]

In [164]:
averages_2015 = result.loc['2015']

In [171]:
averages_2015 = averages_2015.tolist()

In [172]:
averages_SSc1 = result['SSc1']

In [178]:
result['SSc1'].tolist()

[0.84782608695652173,
 0.82983193277310929,
 0.84183006535947713,
 0.8952941176470588,
 0.88167938931297707]

In [190]:
result.drop('Total Points', axis=1, inplace=True)

In [185]:
columns_all = result.columns

In [205]:
ss_2009 = ['SSc1', 'SSc2', 'SSc3', 'SSc4.1', 'SSc4.2', 'SSc4.3', 'SSc4.4',
       'SSc5.1', 'SSc5.2', 'SSc6.1', 'SSc6.2', 'SSc7.1', 'SSc7.2', 'SSc8']

In [206]:
we_2009 = ['WEc1', 'WEc2', 'WEc3']

In [207]:
ea_2009 = ['EAc1', 'EAc2', 'EAc3', 'EAc4', 'EAc5', 'EAc6']

In [208]:
mr_2009 = ['MRc1.1', 'MRc1.2', 'MRc2', 'MRc3', 'MRc4', 'MRc5', 'MRc6', 'MRc7']

In [209]:
eq_2009 = ['EQc1', 'EQc2', 'EQc3.1', 'EQc3.2', 'EQc4.1', 'EQc4.2', 'EQc4.3',
       'EQc4.4', 'EQc5', 'EQc6.1', 'EQc6.2', 'EQc7.1', 'EQc7.2', 'EQc8.1',
       'EQc8.2']

In [218]:
id_and_extra_2009 = ['IDc1', 'IDc2', 'Extra 1', 'Extra 2', 'Extra 3', 'Extra 4']

In [198]:

import plotly.plotly as py
from plotly.graph_objs import *
years = [2011, 2012, 2013, 2014, 2015]

trace1 = Bar(
    x=years,
    y=result['SSc1'].tolist(),
    fill='tozeroy'
)
trace2 = Scatter(
    x=years,
    y=
    fill='tonexty'
)
trace3 = Scatter(
    x=years,
    y=result['SSc3'].tolist(),
    fill='tonexty'
)
trace4 = Scatter(
    x=years,
    y=result['SSc4.1'].tolist(),
    fill='tonexty'
)
data = Data([trace1])



#data = Data([Scatter(x=years, y=result[score_point].tolist(), fill='tonexty') for score_point in ['Extra 1', 'Extra 2', 'Extra 3', 'Extra 4']])
py.iplot(data, filename='LEED score averages per year')

SyntaxError: non-keyword arg after keyword arg (<ipython-input-198-7347b52a3479>, line 8)

In [244]:

# Data([Bar(x=years, y=result[score_point].tolist(), name=score_point) for score_point in result.columns])
data = Data([Bar(x=years, y=result[score_point].tolist(), name=score_point) for score_point in ss_2009])


layout = Layout(
    barmode='stack',
    xaxis=XAxis(
        title='Year',
        titlefont=Font(
            family='Helvetica, monospace',
            size=18,
            color='#7f7f7f'
        )
    ),
    yaxis=YAxis(
        title='Score',
        titlefont=Font(
            family='Helvetica, monospace',
            size=18,
            color='#7f7f7f'
        )
    )
)
fig = Figure(data=data, layout=layout)
fig['layout'].update(
    autosize=False,  # allow custom size (set already, only for completeness here)
    width=960,    # link new width value
    height=453,  # link new height value
#     font=Font(size=7),       # update global font size
#     titlefont=Font(size=12)  #  as well as the title font size 
)
layout.update(margin=Margin(l=50,r=50,b=50,t=50,pad=0,autoexpand=True))
py.iplot(fig, filename='ss_2009_stacked-bar')

Description: Average points achieved for LEED version 2009 Sustainable Sites credits by year.  
Intersting Findings: Performance on these credits are relatively stable on the NC-2009 version.  The dip in years 2012 and 2013 are mostly due to SSc4.1 - Alternative Transportation—Public Transportation Access.  
Further analysis: How to encourage completion of SSc5.1 - Site Development (Protect or Restore Habitat) and SSc3 - Brownfield Redevelopment.

In [243]:
data = Data([Bar(x=years, y=result[score_point].tolist(), name=score_point) for score_point in result.columns])

layout = Layout(
    barmode='stack',
    xaxis=XAxis(
        title='Year',
        titlefont=Font(
            family='Helvetica, monospace',
            size=18,
            color='#7f7f7f'
        )
    ),
    yaxis=YAxis(
        title='Score',
        titlefont=Font(
            family='Helvetica, monospace',
            size=18,
            color='#7f7f7f'
        )
    )
)
fig = Figure(data=data, layout=layout)
fig['layout'].update(
    autosize=False,  # allow custom size (set already, only for completeness here)
    width=960,    # link new width value
    height=453,  # link new height value
#     font=Font(size=7),       # update global font size
#     titlefont=Font(size=12)  #  as well as the title font size 
)
layout.update(margin=Margin(l=50,r=50,b=50,t=50,pad=0,autoexpand=True))
py.iplot(fig, filename='total-2009-stacked-bar')

Description: Average points achieved for LEED version 2009 on all credits.  
Interesting Findings: Overall performance on the public scored projects has declined from 57.25 to 54.45 since 2011.  Much of this underperformance can be traced to a decline in the average scores for EAc1 and EAc2.  
Further analysis: Has the ratio of Platinum, Gold, Silver and Certified projects changed?  If so, what is the driver of the shift to lower LEED certified status?  If not, why are scores declining overall?

In [242]:

# Data([Bar(x=years, y=result[score_point].tolist(), name=score_point) for score_point in result.columns])
data = Data([Bar(x=years, y=result[score_point].tolist(), name=score_point) for score_point in eq_2009])

layout = Layout(
    barmode='stack',
    xaxis=XAxis(
        title='Year',
        titlefont=Font(
            family='Helvetica, monospace',
            size=18,
            color='#7f7f7f'
        )
    ),
    yaxis=YAxis(
        title='Score',
        titlefont=Font(
            family='Helvetica, monospace',
            size=18,
            color='#7f7f7f'
        )
    )
)
fig = Figure(data=data, layout=layout)
fig['layout'].update(
    autosize=False,  # allow custom size (set already, only for completeness here)
    width=960,    # link new width value
    height=453,  # link new height value
#     font=Font(size=7),       # update global font size
#     titlefont=Font(size=12)  #  as well as the title font size 
)
layout.update(margin=Margin(l=50,r=50,b=50,t=50,pad=0,autoexpand=True))
py.iplot(fig, filename='eq-2009-stacked-bar')

Description: Average points achieved for LEED version 2009 - Indoor Environmental Quality credits.  
Interesting Findings: Scores in Indoor Environmental Quality have descreased since 2012, EQc4.4 - Low-Emitting Materials (Composite Wood and Agrifiber Products) has contributed most to the decline.  
Further analysis: Have costs in this space increased above inflation?

In [234]:

# Data([Bar(x=years, y=result[score_point].tolist(), name=score_point) for score_point in result.columns])
data = Data([Bar(x=years, y=result[score_point].tolist(), name=score_point) for score_point in we_2009])

layout = Layout(
    barmode='stack',
    xaxis=XAxis(
        title='Year',
        titlefont=Font(
            family='Helvetica, monospace',
            size=18,
            color='#7f7f7f'
        )
    ),
    yaxis=YAxis(
        title='Score',
        titlefont=Font(
            family='Helvetica, monospace',
            size=18,
            color='#7f7f7f'
        )
    )
)

fig = Figure(data=data, layout=layout)
fig['layout'].update(
    autosize=False,  # allow custom size (set already, only for completeness here)
    width=960,    # link new width value
    height=500,  # link new height value
#     font=Font(size=7),       # update global font size
#     titlefont=Font(size=12)  #  as well as the title font size 
)
py.iplot(fig, filename='we-2009-stacked-bar')

Description: Average points achieved for LEED version 2009 - Water Efficiency credits.    
Interesting Findings: Scores in WEc1 - Water Efficient Landscaping and WEc2 -  Innovative Wastewater Technologies have experienced declines, even as water use and drought has been a major policy concern.   
Further analysis: How can LEED encourage increased performance in this area?  Can LEED take the lead in pushing for new technologies and adoption of water efficient techniques?  If WEc2 is rarely achieved, how can this point be changed or encouraged.  Is the cost for achieving this credit too expensive?

In [239]:

# Data([Bar(x=years, y=result[score_point].tolist(), name=score_point) for score_point in result.columns])
data = Data([Bar(x=years, y=result[score_point].tolist(), name=score_point) for score_point in ea_2009])

layout = Layout(
    barmode='stack',
    xaxis=XAxis(
        title='Year',
        titlefont=Font(
            family='Helvetica, monospace',
            size=18,
            color='#7f7f7f'
        )
    ),
    yaxis=YAxis(
        title='Score',
        titlefont=Font(
            family='Helvetica, monospace',
            size=18,
            color='#7f7f7f'
        )
    )
)
fig = Figure(data=data, layout=layout)
fig['layout'].update(
    autosize=False,  # allow custom size (set already, only for completeness here)
    width=960,    # link new width value
    height=453,  # link new height value
#     font=Font(size=7),       # update global font size
#     titlefont=Font(size=12)  #  as well as the title font size 
)
layout.update(margin=Margin(l=50,r=50,b=50,t=50,pad=0,autoexpand=True))
py.iplot(fig, filename='ea-2009-stacked-bar')

Description: Average points achieved for LEED version 2009 - Energy and Atmosphere credits.    
Interesting Findings: Scores in EAc1 - Optimize Energy Performance have declined, on average, from 9.49 to 7.96 (out of 19 possible) and scores on EAc2 - On-site Renewable Energy have declined from 1.70 to .87 (out of 7 possible).   
Further analysis: How can LEED encourage increased performance in this area?  These credits are extremely important to incentivize, since they are the most directly related to greenhouse emmissions.  Should certification level be linked more directly to these credits?

In [240]:

# Data([Bar(x=years, y=result[score_point].tolist(), name=score_point) for score_point in result.columns])
data = Data([Bar(x=years, y=result[score_point].tolist(), name=score_point) for score_point in mr_2009])

layout = Layout(
    barmode='stack',
    xaxis=XAxis(
        title='Year',
        titlefont=Font(
            family='Helvetica, monospace',
            size=18,
            color='#7f7f7f'
        )
    ),
    yaxis=YAxis(
        title='Score',
        titlefont=Font(
            family='Helvetica, monospace',
            size=18,
            color='#7f7f7f'
        )
    )
)
fig = Figure(data=data, layout=layout)
fig['layout'].update(
    autosize=False,  # allow custom size (set already, only for completeness here)
    width=960,    # link new width value
    height=453,  # link new height value
#     font=Font(size=7),       # update global font size
#     titlefont=Font(size=12)  #  as well as the title font size 
)
layout.update(margin=Margin(l=50,r=50,b=50,t=50,pad=0,autoexpand=True))
py.iplot(fig, filename='mr-2009-stacked-bar')

Description: Average points achieved for LEED version 2009 - Materials and Resources credits.    
Interesting Findings: Scores in MRc 1.2 - Building Reuse (Maintain Existing Interior Nonstructural Elements), MRc3 - Materials Reuse, and MRc6 -  Rapidly Renewable Materials are almost never achieved.  
Further analysis: When score points are not achieved, should they be reanalyzed?  Do they not apply to all projects (perhaps for MRc3)?  Are they too expensive relative to the point total?  How could they be incentivized in the next version?

In [241]:

# Data([Bar(x=years, y=result[score_point].tolist(), name=score_point) for score_point in result.columns])
data = Data([Bar(x=years, y=result[score_point].tolist(), name=score_point) for score_point in id_and_extra_2009])

layout = Layout(
    barmode='stack',
    xaxis=XAxis(
        title='Year',
        titlefont=Font(
            family='Helvetica, monospace',
            size=18,
            color='#7f7f7f'
        )
    ),
    yaxis=YAxis(
        title='Score',
        titlefont=Font(
            family='Helvetica, monospace',
            size=18,
            color='#7f7f7f'
        )
    )
)
fig = Figure(data=data, layout=layout)
fig['layout'].update(
    autosize=False,  # allow custom size (set already, only for completeness here)
    width=960,    # link new width value
    height=453,  # link new height value
#     font=Font(size=7),       # update global font size
#     titlefont=Font(size=12)  #  as well as the title font size 
)
layout.update(margin=Margin(l=50,r=50,b=50,t=50,pad=0,autoexpand=True))
py.iplot(fig, filename='regional-id-2009-stacked-bar')

Description: Average points achieved for LEED version 2009 - Innovation in Design and Regional Priority Credits (referenced as Extra).  
Interesting Findings: Scores are relatively stable in these categories.  Almost every project gets a credit for using a LEED Certified professional (IDc2).  Perhaps this credit is unnecessary, since having a LEED professional seems to be essentially required to achieve certification?  
Further analysis: How are the other Innovation in Design credits awarded?  Is there variablity across projects or do most score near the average?  What is the average completion of the different regional priority credits?  